src/shared/sgxlkl_enclave_config.c

```diff
@@ -0,0 +1,560 @@
+            JBOOL("mounts.create", MOUNT()->create);
+            JBOOL("mounts.fresh_key", MOUNT()->fresh_key);
+            JHEXBUF("mounts.key", MOUNT()->key);
+            JSTRING("mounts.key_id", MOUNT()->key_id);
+            if (MATCH("mounts.destination"))
+            {
+                SEEN(parser);
+                size_t len = strlen(un->string);
+                if (len > SGXLKL_DISK_MNT_MAX_PATH_LEN)
+                    FAIL("invalid length of 'mnt' for disk %d\n", i);
+                memcpy(MOUNT()->destination, un->string, len + 1);
+                return JSON_OK;
+            }
+            JBOOL("mounts.readonly", MOUNT()->readonly);
+            JSTRING("mounts.roothash", MOUNT()->roothash);
+            JU64("mounts.roothash_offset", MOUNT()->roothash_offset);
+            JU64("mounts.size", MOUNT()->size);
+
+            sgxlkl_image_sizes_config_t* sizes = &cfg->image_sizes;
+            JU64("image_sizes.num_heap_pages", sizes->num_heap_pages);
+            JU64("image_sizes.num_stack_pages", sizes->num_stack_pages);
+
+            // Else element is unknown. We ignore this to allow newer
+            // launchers to support options that older enclave images
+            // don't know about.
+            WARN("Ignoring unknown json element '%s'.\n", make_path(parser));
+        }
+    }
+
+    return JSON_OK;
+}
+
+void check_config(const sgxlkl_enclave_config_t* cfg)
+{
+#define CC(C, M) \
+    if (C)       \
+        FAIL("rejecting enclave configuration: " M "\n");
+
+    CC(cfg->args == NULL, "missing args");
+    CC(cfg->num_args == 0, "num_args == 0");
+    CC(cfg->ethreads > MAX_SGXLKL_ETHREADS, "too many ethreads");
+    CC(cfg->max_user_threads > MAX_SGXLKL_MAX_USER_THREADS,
+       "max_user_threads too large");
+
+    // These are cast to (signed) int later.
+    CC(cfg->tap_mtu > INT32_MAX, "tap_mtu out of range");
+    CC(cfg->num_args > INT32_MAX, "size of args out of range");
+    CC(cfg->num_env > INT32_MAX, "size of env out of range");
+    CC(cfg->num_auxv > INT32_MAX, "size of auxv out of range");
+    CC(cfg->num_host_import_env > INT32_MAX,
+       "size of host_import_env out of range");
+}
+
+void check_required_elements(string_list_t* seen)
+{
+    const char* required[] = {"args"};
+    const size_t num_required = sizeof(required) / sizeof(char*);
+    for (size_t i = 0; i < num_required; i++)
+    {
+        const char* r = required[i];
+        if (!string_list_contains(seen, r))
+            FAIL("config does not contain required element '%s'\n", r);
+    }
+}
+
+int sgxlkl_read_enclave_config(
+    const char* from,
+    sgxlkl_enclave_config_t* to,
+    bool enforce_format)
+{
+    // Catch modifications to sgxlkl_enclave_config_t early. If this fails,
+    // the code above/below needs adjusting for the added/removed settings.
+    _Static_assert(
+        sizeof(sgxlkl_enclave_config_t) == 496,
+        "sgxlkl_enclave_config_t size has changed");
```

Comment:
This doesn't output anything, it's a static check, not a runtime check, so the compiler throws this error.

Reply:
However you want to solve it, the current size should be output as well. If this can't be done with a compiler message then we need to find another way. Devs will not start computing the new number manually.

Reply:
I don't compute that number manually either. Intellisense shows it when you hover over it (works best if you move the `sizeof` out of the `_Static_assert`. If you find a better way, I'm happy to change it though.